sdk/src/context.rs

```diff
@@ -74,44 +78,32 @@ where
 {
     pub fn new(
         anchor_context: Context<'a, 'b, 'c, 'info, T>,
-        inputs: Vec<Vec<u8>>,
-        merkle_context: PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: PackedAddressMerkleContext,
-        address_merkle_tree_root_index: u16,
+        inputs: LightInstructionInputs,
     ) -> Result<Self> {
-        let light_accounts = U::try_light_accounts(
-            inputs,
-            merkle_context,
-            merkle_tree_root_index,
-            address_merkle_context,
-            address_merkle_tree_root_index,
-            anchor_context.remaining_accounts,
-        )?;
+        let light_accounts = U::try_light_accounts(&inputs)?;
         Ok(Self {
             anchor_context,
             light_accounts,
+            inputs,
         })
     }
 
-    pub fn verify(&mut self, proof: CompressedProof) -> Result<()> {
+    pub fn verify(&mut self) -> Result<()> {
```

Comment:
If this was a function as follows we could probably reuse it in a rust native sdk: `pub fn verify<T: LightAccountsTrait>(accounts: T, proof: Option<ProofRpcResult>, compressed_accounts: Option<Vec<PackedCompressedAccountWithMerkleContext>>, new_addresses: Option<Vec<PackedNewAddressParams>>) -> Result<(), SdkError>`
- note that Result should be the anchor result if we want to reuse it in a rust native sdk

Reply:
note I am not 100% sure this will work

Reply:
I think it should be possible to remove anchor as a dependency since all the underlying functions just access `LightTraits` of `ctx.accounts`

Reply:
Good idea, your proposed signature would allow to verify just a subset of accounts, which is what you were talking about.